packages/trader/src/Modules/Contract/Containers/contract-replay.tsx

```diff
@@ -0,0 +1,384 @@
+import classNames from 'classnames';
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import {
+    DesktopWrapper,
+    Div100vhContainer,
+    MobileWrapper,
+    PageOverlay,
+    SwipeableWrapper,
+    FadeWrapper,
+    usePrevious,
+} from '@deriv/components';
+import {
+    getContractTypeFeatureFlag,
+    getDurationPeriod,
+    getDurationUnitText,
+    getEndTime,
+    getPlatformRedirect,
+    isAccumulatorContract,
+    isDesktop,
+    isEmptyObject,
+    isHighLow,
+    isMobile,
+    isMultiplierContract,
+    isTurbosContract,
+    isVanillaContract,
+    isSmartTraderContract,
+    urlFor,
+} from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import { useFeatureFlags } from '@deriv/hooks';
+import ChartLoader from 'App/Components/Elements/chart-loader';
+import ContractDrawer from 'App/Components/Elements/ContractDrawer';
+import UnsupportedContractModal from 'App/Components/Elements/Modals/UnsupportedContractModal';
+import SmartChartSwitcher from '../../Trading/Containers/smart-chart-switcher.jsx';
+import { ChartBottomWidgets, ChartTopWidgets, DigitsWidget, InfoBoxWidget } from './contract-replay-widget.jsx';
+import ChartMarker from 'Modules/SmartChart/Components/Markers/marker.jsx';
+import DelayedAccuBarriersMarker from 'Modules/SmartChart/Components/Markers/delayed-accu-barriers-marker';
+import allMarkers from 'Modules/SmartChart/Components/all-markers.jsx';
+import ChartMarkerBeta from 'Modules/SmartChartBeta/Components/Markers/marker.jsx';
+import { observer, useStore } from '@deriv/stores';
+import { useTraderStore } from 'Stores/useTraderStores';
+
+type TLocation = Location & { state: { from_table_row: boolean } | Record<string, never> };
```

Comment:
```suggestion
type TLocationState = {
   from_table_row: boolean
}

const ContractReplay = observer(({ contract_id }: { contract_id: number }) => {
    const { state } = useLocation<TLocationState>();
    
```


I would suggest creating a separate type for the location state and using it as a hook

Reply:
Great, thank you)